[ADD] rental: new product type page (#17899)
odoo/documentation: content/applications/sales/rental/configure_products/product_type.rst

.. _configure-rental-product: https://youtu.be/CE-SahTUC9A?si=Gae5tLAZk6wt70TW

===================
Rental product type
===================

Odoo allows for two :ref:`product types <inventory/product_management/goods-or-services>` when

> configuring a product in the Odoo ecosystem: Goods or Services.

Reviewer: I know what you're saying but it might be misleading the viewer to think that there are only 2 product types in odoo, but there are others. but, they're unrelated to Rental, so maybe we can specify that!

```suggestion
Odoo allows for two :ref:`product types <inventory/product_management/goods-or-services>` when
configuring a rental product: Goods or Services.
```

returned to the company.

A rental *Service* can be labor or something non-physical, such as installation, photography, or a
deposit. It can also be a physical item that never leaves the company’s stock, such as a hotel room,
work station, or storage unit.

The following sections detail basic settings and app integration configurations for Goods and
Services.

Settings
========

.. important::
   For the :guilabel:`Default Padding time` and :guilabel:`Rental Transfers` settings, the
   **Inventory** app must be installed. For the *Rent Online section* settings, the **eCommerce**
   module must be installed.

To configure default settings on rental products, navigate to :menuselection:`Rental app -->
Configuration --> Settings`.

.. image:: product_type/rental-settings.png
   :alt: The settings page of the Rental app with all the default values.

In the *Rental* section, the *Default Delay Costs* subsection is used to set up late fees. These are
the base settings that come with the **Rental** app installation. To apply delay costs and times to
rental products, fill in the:

- :guilabel:`Per Hour`: The late fee amount is applied each hour the rental product is past due.

> - :guilabel:`Per Day``: The late fee amount is applied each day the rental product is past due.

Reviewer: ```suggestion
- :guilabel:`Per Day`: The late fee amount is applied each day the rental product is past due.
```

- :guilabel:`Apply after`: The number of hours necessary to trigger a late fee.
- :guilabel:`Product`: Limit the default delay costs setting to one product from the field’s

>   drop-down list. Leave the Product field blank if the defaults apply to all products.

Reviewer: I'm thinking we format "Product" here with guilabel since it's on the UI (:

If the **Inventory** app is installed, the *Default Padding Time* section displays. The time

> inputted into :guilabel:`Padding` field blocks a rental product from being available for

Reviewer: ```suggestion
inputted into the :guilabel:`Padding` field blocks a rental product from being available for
```

reservations. The setting is set to an hourly unit.

.. note::
   For more control, configure the costs for late returns and padding time in the :guilabel:`Hourly
   Fine`, :guilabel:`Daily Fine`, and :guilabel:`Reserve product` fields on the product form.

The :guilabel:`Rental Transfers` checkbox enables automatic creation of delivery and return receipts
for a rental product. This feature also needs the **Inventory** app to be installed.

.. note::
   The **Inventory** app automatically creates an internal default location once the
   :guilabel:`Rental Transfers` feature is enabled. Odoo uses the new default location,
   `Customer/Rental`, to track products during the rental period (moving them from Stock to

>    Customer/Rental upon rental, and back upon return). **Do not** modify to avoid corrupting

Reviewer: ```suggestion
   Customer/Rental upon rental, and back upon return). **Do not** modify this location to avoid corrupting
```

   inventory tracking.

If the **eCommerce** module is installed, the *Rent Online* section is available to configure. The
*Minimal time of rental* section contains:

- :guilabel:`Minimal Rental Duration`: The minimum duration of time a rental product must be booked.

>   The available units of time are: Hours, Days, Weeks, and Months.

Reviewer: are these default units in Odoo? i think we should italicize them! or we just leave the terms lowercase if you're referring to these units of broadly

Author: These are fields within the Rent Online section of the Settings page. I used guilabales for the options and kept the label capitalization.

The **Rental** app relies on additional Odoo app integrations to expand its settings and product
configurations. The following apps are essential for workflow efficiency and automation when
creating a product and rental order:

- **Sales**: Enables online payments and signatures, uses quotation templates, variants, and PDF
  Quote Builder.
- **Sign**: Allows uploading and customization of various rental and service agreements. These
  documents are used to facilitate the *Request Signature* feature.
- **Planning**: Integrates with the **Rental** app to automatically match service products with
  employees or materials based on availability. This setting is configured on the product form.
- **Project**: Integrates with the **Rental** app to automatically creation of a project, a task, or
  both when a rental quote that includes the configured product is confirmed. This setting is
  configured on the product form.
- **Inventory**: Enables warehouse delivery and return receipts, product tracking options, variants,
  and product downtime.
- **eCommerce**: Allows product configuration for the online shop. This setting is configured on the
  product form.

.. seealso::

> 

Reviewer: extra line
```suggestion
```

   - :doc:`../../sales`
   - :doc:`../../sales/sales_quotations/pdf_quote_builder`
   - :doc:`../../../services/planning`
   - :doc:`../../../productivity/sign`
   - :doc:`../../../services/project`
   - :doc:`../../../inventory_and_mrp/inventory`
   - :doc:`../../../websites/ecommerce`

Search for rental products
==========================

To view all products available for rent in the database, navigate to :menuselection:`Rentals app -->
Products`. By default, the :guilabel:`Rental` filter appears in the search bar, and the view is
Kanban. Remove the filter, then click the search bar. From the preset filters, select
:guilabel:`Goods`, :guilabel:`Services`, or both.

All the selected options appear as Kanban cards. For Goods, the card displays the name, rental rate,

> and amount on hand. For Services, the card displays the name, the number of variants if configured,

Reviewer: ```suggestion
All the selected options appear as Kanban cards. For *Goods*, the card displays the name, rental rate,
and amount on hand. For *Services*, the card displays the name, the number of variants if configured,
```

and the rental price.

.. seealso::

   - :doc:`products`
   - :doc:`service_products`
   - `Tutorial: Configuring a rental product <configure-rental-product_>`_

Rental periods
==============

A rental period is the unit of time used to calculate the rental rate. This setting is configured on
the *Rental prices* tab on the product form. To edit a default rental rate, navigate to
:menuselection:`Rental app --> Configuration --> Rental Periods`.

Select a rental period to open the *Period* form. With the exception of the :guilabel:`Nightly`
rental period, all period forms have the following fields:

- :guilabel:`Name`: A short description or title for the period.
- :guilabel:`Duration`: The minimum amount of time before the rule is applied. Set to `0` to
  represent a fixed rental price.
- :guilabel:`Unit`: A drop-down list of time periods the shortest being Hours and the longest is
  Years.

To create a new rental period, click :guilabel:`New`.

Nightly rental period
---------------------

The *Nightly* rental period form replaces the :guilabel:`Duration` field with two new fields

> :guilabel:`Check-in` and :guilabel:`Check-out`. Both fields use a 24 hour clock to set the time.

Reviewer: ```suggestion
:guilabel:`Check-in` and :guilabel:`Check-out`. Both fields use a 24-hour clock to set the time.
```

.. _configure-rental-product: https://youtu.be/CE-SahTUC9A?si=Gae5tLAZk6wt70TW

===================
Rental product type
===================

Odoo allows for two :ref:`product types <inventory/product_management/goods-or-services>` when
configuring a product in the Odoo ecosystem. Deciding to configure a product as a Goods or a Service
in the **Rental** app depends on a few factors:

- Does the rental product require stock movement?

> - Are shifts necessary for the rental product?

Reviewer 2: I feel like these psuedo-rhetorical questions don't quite jive with our style guide, but I concede that that's a personal preference.

> A rental *Goods* product is a physical item, such as computers, vehicles, bikes, cameras, or

Reviewer 2: Subject-verb agreement (or technically, main clause-dependent clause or something).

clothes. These Goods are removed from the company’s stock, delivered to the customer, and then
returned to the company.

A rental *Service* product can be a physical item that never leaves the company’s stock, such as a
hotel room, work station, or storage unit. A Service can also be labor or something non-physical,

> such as installation, photography, or a deposit.

Reviewer 2: Maybe swap the order here? Feels like a service should start with an intangible item, and then you can mention that it can also include physical goods that get returned.

The following sections detail basic settings and app integration configurations for Goods and
Services.

Settings
========

.. important::
   For the :guilabel:`Default Padding time` and :guilabel:`Rental Transfers` settings, the
   **Inventory** app must be installed. For the *Rent Online section* settings, the **eCommerce**
   module must be installed.

> To configure default settings on rental products, navigate to the :menuselection:`Rental app -->

Reviewer 2: ```suggestion
To configure default settings on rental products, navigate to :menuselection:`Rental app -->
```

Configuration --> Settings`.

.. image:: product_type/rental-settings.png
   :alt: The settings page of the Rental app with all the default values.

In the *Rental* section, the *Default Delay Costs* subsection is used to set up late fees. These are
the base settings that come with the **Rental** app installation. To apply delay costs and times to

> all rental products, or for one product, fill in the:

Reviewer 2: ```suggestion
the base settings that come with the **Rental** app installation. To apply delay costs and times to
rental products, fill in the:
```

Maybe unnecessary to specify one or all? Intuitive in the UI?

- :guilabel:`Per Hour`: The late fee amount is applied each hour the rental product is past due.
- :guilabel:`Per Day``: The late fee amount is applied each day the rental product is past due.
- :guilabel:`Apply after`: The number of hours necessary to trigger a late fee.
- :guilabel:`Product`: Limit the default delay costs setting to one product from the field’s
  drop-down list. Leave the Product field blank if the defaults apply to all products.

If the **Inventory** app is installed, the *Default Padding Time* section displays. The time
inputted into :guilabel:`Padding` field blocks a rental product from being available for
reservations. The setting is set to an hourly unit.

.. note::
   For more control, configure the costs for late returns and padding time in the :guilabel:`Hourly
   Fine`, :guilabel:`Daily Fine`, and :guilabel:`Reserve product` fields on the product form.

The :guilabel:`Rental Transfers` checkbox enables automatic creation of delivery and return receipts

> for a rental product. This feature is also needs the **Inventory** app to be installed.

Reviewer 2: ```suggestion
for a rental product. This feature also needs the **Inventory** app to be installed.
```

App integration configuration
=============================

The **Rental** app relies on additional Odoo app integrations to expand its settings and product
configurations. The following apps are essential for workflow efficiency and automation when
creating a product and rental order:

- **Sales** app: Enables online payments and signatures, uses quotation templates, variants, and PDF
  Quote Builder.
- **Sign app**: Allows uploading and customization of various rental and service agreements. These
  documents are used to facilitate the *Request Signature* feature.
- **Planning** app: Integrates with the **Rental app** to automatically match service products with
  employees or materials based on availability. This setting is configured on the product form.
- **Project** app: Enables automatic creation of a project, a task, or both when a rental quote that
  includes the configured product is confirmed. This setting is configured on the product form.
- **Inventory** app: Enables warehouse delivery and return receipts, product tracking options,
  variants, and product downtime.
- **eCommerce** app: Enables product configuration for the online shop. This setting is configured

>   on the product form.

Reviewer 2: Maybe trim "app" from these bullets? You introduce in the paragraph above that these are all apps.

Also, so many of these have "Enables" as the opening verb. Any way to get "Allows" and "Integrates" to be "Enables" as well? Might just be me being neurotic.  xP

.. seealso::

   - :doc:`../../sales`
   - :doc:`../../sales/sales_quotations/pdf_quote_builder`
   - :doc:`../../../services/planning`
   - :doc:`../../../productivity/sign`
   - :doc:`../../../services/project`
   - :doc:`../../../inventory_and_mrp/inventory`
   - :doc:`../../../websites/ecommerce`

Search for rental products
==========================

> To view all products available for rent in the database, navigate to the :menuselection:`Rentals app

Reviewer 2: ```suggestion
To view all products available for rent in the database, navigate to :menuselection:`Rentals app
```

--> Products`. By default, the :guilabel:`Rental` filter appears in the search bar, and the view is
Kanban. Remove the filter, then click the search bar. From the preset filters, select
:guilabel:`Goods`, :guilabel:`Services`, or both. All the selected options appear as Kanban cards.

> For Goods, the card displays the name, rental rate, and amount on hand. For Services, the card

Reviewer 2: Feels like these lines could be one paragraph since they're still one thought. Stylistic thing, though, so take it or leave it as you see fit.

displays the name, the number of variants if configured, and the rental price.

.. seealso::

   - :doc:`products`
   - :doc:`service_products`
   - `Tutorial: Configuring a rental product <configure-rental-product_>`_

Rental periods
==============

A rental period is the unit of time used to calculate the rental rate. This setting is configured on

> the *Rental prices* tab on the product form. To edit a default rental rate, navigate to the

Reviewer 2: ```suggestion
the *Rental prices* tab on the product form. To edit a default rental rate, navigate to
```

:menuselection:`Rental app --> Configuration --> Rental Periods`.

Select a rental period to open the *Period* form. With the exception of the :guilabel:`Nightly`
rental period, all period forms have the following fields:

> - :guilabel:`Name`: Short description or title for the period.

Reviewer 2: ```suggestion
- :guilabel:`Name`: A short description or title for the period.
```

- :guilabel:`Duration`: The minimum amount of time before the rule is applied. Set to `0` to
  represent a fixed rental price.
- :guilabel:`Unit`: A drop-down list of time periods the shortest being Hours and the longest is
  Years.

> To create a new rental period click :guilabel:`New`.

Reviewer 2: ```suggestion
To create a new rental period, click :guilabel:`New`.
```
